### DISCUSSION TITLE
haserl CGI Error: Error passing the lua library to the lua vm: luascript.lua: bad binary format (lua_Integer size mismatch)

### DISCUSSION THREAD

**Discussion by kainers1** [Freetz-NG/freetz-ng#1008] [Allgemein]
bin auf die 7590ax umgestiegen und habe dabei auch freetz-ng neu gebaut. Was ich nicht hinbekomme, sind lua scripte via inetd ([anleitung](https://github.com/Freetz-NG/freetz-ng/blob/master/docs/make/haserl.md))

Fehlermeldung ist:
```
haserl CGI Error
Error passing the lua library to the lua vm: luascript.lua: bad binary format (lua_Integer size mismatch)
```

Jemand eine Idee dazu?

[config.txt](https://github.com/user-attachments/files/16042356/config.txt)

**Comment by kainers1** [comment]
Solved in #662

**Comment by fda77** [comment]
Hmm, klär mich doch mal auf. Evtl hattest du vorher eine mips und nun eine arm box? Und der bytecode (oder wie das bei lua heisst) passte nicht? Dann müsste man den Patch noch arch-abhängig machen...

**Reply by kainers1** [reply]
bin von einer 7590 auf eine 7590ax gewechselt. Habe mich nicht gleich an mein altes Ticket erinnert und einige Zeit gebraucht, bis ich herausfand dass `FREETZ_PACKAGE_LUA_FLOAT=y` benötigt wird. Das heißt (für die 7590ax) wenn `(haserl) with lua Support `
 ausgewählt wird, wird muss auch `use float instead double` ausgewählt werden.

**Comment by fda77** [comment]
Du hast also bei packages lua aktiviert, aber nicht gleich bemerkt dass lua noch eine lib hat die dann noch diese option hat...?
Dann könnte man beim package noch eine Option dazu machen die nur die Lib-Option wählt, falls das nächstes mal helfen würde. 
EDIT
Allerdings würde es dann nicht reichen die Option beim Package zu deaktivieren, da sie dann bei der Lib an bliebe

**Reply by kainers1** [reply]
> Du hast also bei packages lua aktiviert, aber nicht gleich bemerkt dass lua noch eine lib hat die dann noch diese option hat...?

ja, weil im beginner-mode die option nicht zugänglich ist. Ob eine Änderung sinnvoll ist, kann ich nicht wirklich beurteilen.

**Reply by fda77** [reply]
Diese elenden Benutzer modi. Ich würde die ja gerne komplett rauswerfen, scheinen aber beliebt zu sein. Als ich die von  3 auf 2 Stufen reduziert hab gab es schon einige Beschwerden
